### PR TITLE
transpile self: Add more languages

### PR DESCRIPTION
**Pull request by jayvdb**
Simplify a raise in cli.py to avoid tripping up
transpilers.

Raise `NotImplementedError` in py2many to avoid
suppressing `Exception`.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -50,11 +50,15 @@ jobs:
           pip3 --version
           pip3 install setuptools tox
 
-      - name: Install Julia dependencies for ${{ runner.os }}.
+      - name: Install Julia packages
```
Comment (adsharma) [inline .github/workflows/main.yml:53]: Ah makes sense. They were not all about Julia.

File: py2many/clike.py [hunk 2]
```diff
@@ -225,13 +225,13 @@ def _typename_from_annotation(self, node, attr="annotation") -> str:
                 node.container_type = type_node.container_type
                 return self._visit_container_type(type_node.container_type)
             if typename is None:
-                raise Exception(f"Could not infer: {type_node}")
+                raise NotImplementedError(f"Could not infer: {type_node}")
```
Comment (adsharma) [inline py2many/clike.py:228]: This one could be `TypeError` with `node.lineno`. I'm sure there is a format we can report the errors in that allows IDEs and other tools to implement "click to go to the error". But that can happen in the code that catches the exception.

Comment (jayvdb) [reply py2many/clike.py:228]: It is the transpiler logic is not working properly, iirc this was Julia only which was triggering this for lots of pyjl/ py2many/ inputs.

Implementing a custom subclass of `NotImplementedError` which has extra attributes for context would be useful, but is beyond the planned scope of this PR.  When building that, the implementation should also consume that extra context information in a meaningful way in order to ensure it is designed correctly.

Comment (jayvdb) [reply py2many/clike.py:228]: Julia Call in pyjl/transpiler.py and Go Return in pygo/transpiler.py trigger this.  In both cases `type_node` is "None".

I can see why you think `TypeError` is appropriate, but the logic in those transpilers is not "wrong" - it isn't using the wrong type when calling this function.  The calling logic is just not able to handle more complex scenarios.

The problem with raising normal python exceptions for any of these types of problems is then we need to mask them, and real uses of those exceptions, caused by bad python code, will also be masked, allowing regressions.

In the end, all of these `NotImplementedError` need to be eradicated.  In some cases, like here, they should be replaced with `assert ..` or even `TypeError`.

In addition we should create a special subclass of `NotImplementedError` for each scenario which are not "ever" going to be implemented, such as
- syntax which is not possible in the target language, 
- syntax we dont intend to support for any language, etc

**Comment by jayvdb** [comment]
Windows coverage is now at 89% coverage.

**Comment by jayvdb** [comment]
ping @adsharma , now green

File: tests/test_cli.py [hunk 3]
```diff
@@ -198,46 +201,46 @@ def test_generated(self, case, lang):
...
+                self.assertFalse(proc.returncode)
+
+                if expect_failure:
+                    raise AssertionError(f"{case}{ext} passed unexpectedly")
 
         finally:
             if not self.KEEP_GENERATED:
                 case_output.unlink(missing_ok=True)
-            exe.unlink(missing_ok=True)
+                exe.unlink(missing_ok=True)
```
Comment (adsharma) [inline tests/test_cli.py:243]: So KEEP_GENERATED means keep the generated source as well as the binary.
Sounds good. Wanted to make sure this was intentional.

Comment (jayvdb) [reply tests/test_cli.py:243]: Ya, occasionally I need to run the exe afterwards to see why it is failing.

**Pull request merged**